server/routes.js

```diff
@@ -378,6 +379,29 @@
         }
     });
 
+    var storage = multer.diskStorage({
+        destination: function (req,file, callback){
+            callback(null, String(req.body.upload_path)) 
+        },
+        filename: function(req,file,callback){
+            callback(null, file.originalname)
+        }
+    });
+
+    var upload = multer({storage: storage }).array('file',3);
+
+    router.post('/api/file/upload', function(req,res){
+        upload(req,res,function(err){
+            if(err){
+                res.send("File Upload Failed!");
+                return;
+            }
+            else{
+                res.send("File Uploaded Successfully!");
```

Comment:
in the cases where the msgs are literally a constant. We use HTTP error code. It is usually 400 that stands for bad request etc

See here:
https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html

If you want to send any msg as a tip for the error

```
res.status(400).("Bad request msg here");
```